cmd/prometheus-config-reloader/main.go

```diff
@@ -242,10 +245,47 @@ func createHTTPClient(timeout *time.Duration) http.Client {
 		InsecureSkipVerify: true, // TLS certificate verification is disabled by default.
 	}
 
+	var rt http.RoundTripper = transport
+
+	if podCredentialsUsername != "" || podCredentialsPasswordFile != "" {
+		password, err := os.ReadFile(fmt.Sprintf("%s/password", podCredentialsPasswordFile))
+		if err != nil {
+			fmt.Printf("Failed to read password file: %v\n", err)
+		} else {
+			rt = newBasicAuthRoundTripper(podCredentialsUsername, strings.TrimSpace(string(password)), rt)
```

Comment:
There is an existing basic roundtripper in prometheus common `config.NewBasicAuthRoundTripper()` https://github.com/prometheus/common/blob/main/config/http_config.go#L839 maybe it would be best to reuse it